bin/runtests.php

```diff
@@ -0,0 +1,490 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Formatter\OutputFormatterStyle;
+use Symfony\Component\Console\Input\ArgvInput;
+use Symfony\Component\Console\Input\InputDefinition;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\ConsoleOutput;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Process\Process;
+
+// Setup the runner application
+// ============================
+
+// ensure that we don't timeout
+set_time_limit(0);
+
+// require the autoloader
+$autoloadFile = __DIR__ . '/../vendor/autoload.php';
+if (!file_exists($autoloadFile)) {
+    writeln(sprintf(
+        'Could not find autoload file "%s", make sure you have ' .
+        'installed the projects dependencies (i.e. `composer install`).',
+        $autoloadFile
+    ));
+    exit(255);
+}
+require_once $autoloadFile;
+
+// setup the application - not that we ONLY use the Application class
+// to get the terminal dimensions currently, not for "hosting" any commands
+// etc.
+$application = new Application();
+list($termWidth, $termHeight) = $application->getTerminalDimensions();
+
+$definition = new InputDefinition();
+$definition->addOption(new InputOption('initialize', 'i', InputOption::VALUE_NONE));
+$definition->addOption(new InputOption('flags', 'f', InputOption::VALUE_REQUIRED));
+$definition->addOption(new InputOption('target', 't', InputOption::VALUE_REQUIRED));
+$definition->addOption(new InputOption('all', 'a', InputOption::VALUE_NONE));
+$definition->addOption(new InputOption('jackrabbit-restart', 'r', InputOption::VALUE_NONE));
+$definition->addOption(new InputOption('help', 'h', InputOption::VALUE_NONE));
+$definition->addOption(new InputOption('no-component-tests', 'C', InputOption::VALUE_NONE, null));
+$definition->addOption(new InputOption('no-bundle-tests', 'B', InputOption::VALUE_NONE, null));
+$input = new ArgvInput(null, $definition);
+$output = new ConsoleOutput();
+$output->getFormatter()->setStyle('red', new OutputFormatterStyle('red'));
+$failedTests = new \ArrayObject();
+global $output, $input, $failedTests, $termWidth;
```

Comment:
This isn't a sophisticated script :)

On Fri, Mar 25, 2016 at 05:17:35AM -0700, Daniel Rotter wrote:
>    In [1]bin/runtests.php:
> 
>  > +list($termWidth, $termHeight) = $application->getTerminalDimensions();
>  > +
>  > +$definition = new InputDefinition();
>  > +$definition->addOption(new InputOption('initialize', 'i', InputOption::VALUE_NONE));
>  > +$definition->addOption(new InputOption('flags', 'f', InputOption::VALUE_REQUIRED));
>  > +$definition->addOption(new InputOption('target', 't', InputOption::VALUE_REQUIRED));
>  > +$definition->addOption(new InputOption('all', 'a', InputOption::VALUE_NONE));
>  > +$definition->addOption(new InputOption('jackrabbit-restart', 'r', InputOption::VALUE_NONE));
>  > +$definition->addOption(new InputOption('help', 'h', InputOption::VALUE_NONE));
>  > +$definition->addOption(new InputOption('no-component-tests', 'C', InputOption::VALUE_NONE, null));
>  > +$definition->addOption(new InputOption('no-bundle-tests', 'B', InputOption::VALUE_NONE, null));
>  > +$input = new ArgvInput(null, $definition);
>  > +$output = new ConsoleOutput();
>  > +$output->getFormatter()->setStyle('red', new OutputFormatterStyle('red'));
>  > +$failedTests = new \ArrayObject();
>  > +global $output, $input, $failedTests, $termWidth;
> 
>    Isn't there a better option? Using global is not very sophisticated
>    [2]:confused:
> 
>    —
>    You are receiving this because you authored the thread.
>    Reply to this email directly or [3]view it on GitHub
> 
>    Reverse link: [4]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu-io/sulu/pull/2184#discussion_r57439655
>    3. https://github.com/sulu-io/sulu/pull/2184/files/fe0af7bd20521c6e7cc60d5f84d163d3d6cf1997#r57439655
>    4. https://github.com/sulu-io/sulu/pull/2184/files/fe0af7bd20521c6e7cc60d5f84d163d3d6cf1997#r57439655